adaptive_ai/services/model_selector.py

```
@@ -89,76 +88,93 @@
+                            "easy": easy_threshold,
+                            "medium": medium_threshold,
+                            "hard": hard_threshold,
+                        },
+                    }
+                )
+
+            return results
 
         except Exception as e:
             logger.error(f"Error in model selection: {str(e)}")
             raise ModelSelectionError(f"Failed to select model: {str(e)}")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)